### PR TITLE
build: add lighthouse PWA check before release 

### PR DESCRIPTION
**Pull request by ivelin**
Trying to isolate @sniperadmin 's lhci contribution from #365 into a separate PR so we can expedite merge.

### CODE REVIEW

**Comment by ivelin** [comment]
@sniperadmin 

Trying to keep this PR as small as possible, but it still got to Size M.
LHCI checks lowered to `preset: "lighthouse:no-pwa"` .
Local build tests pass, but Netlify preview fails. Apparently there are differences introduced by netlify. Working on the resolution.

The next step will be to bump LHCI tests one step higher to:
preset: "lighthouse:recommended"

I already know this introduces a bunch of failures, including missing service worker. Let's try to incrementally solve and push updates in bite sized PRs, because each code change could affect UX. Our still low test coverage doesn't give me much confidence with any of these updates, but we have to keep moving forward.

**Comment by sniperadmin** [comment]
@ivelin That's a lot of work,.. 🔥 
I am still studying the changes per commit

**Comment by ivelin** [comment]
> @ivelin That's a lot of work,.. 🔥
> I am still studying the changes per commit

Unfortunately I can't make a lot of these tests locally since they involve github action script behavior. That results in a lot of noisy commits on this PR branch. I'll tag you when I'm done so you can review a stable version.

**Comment by ivelin** [comment]
and now we are at size/L  :/

**Comment by sniperadmin** [comment]
np,.. I know that this PR will have lots of work anyway..
Nice job so far!

**Comment by ivelin** [comment]
OK, I will pause for today, the screen is getting blurry. :)
Still not ready for merge. Can't get past Netlify LHCI failures even at the lowest "no-pwa" test level. 
This is what's left to address at a minimum to be able to merge:
https://github.com/ambianic/ambianic-ui/actions/runs/103070453

Annotations
 Check failure on line 1 in .github

@github-actions
github-actions
/ build

.github#L1
16 results for https://deploy-preview-367--ambianic-pwa-dist.netlify.app/
Report: https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1589328908819-1768.report.html

❌ `font-display` failure for `minScore` assertion (Ensure text remains visible during webfont load: https://web.dev/font-display)
Expected >= 1, but found 0

❌ `is-crawlable` failure for `minScore` assertion (Page is blocked from indexing: https://web.dev/is-crawable)
Expected >= 1, but found 0

❌ `unused-css-rules` failure for `maxLength` assertion (Remove unused CSS: https://web.dev/unused-css-rules)
Expected <= 0, but found 3

❌ `uses-rel-preconnect` failure for `maxLength` assertion (Preconnect to required origins: https://web.dev/uses-rel-preconnect)
Expected <= 0, but found 2

**Comment by ivelin** [comment]
Turning off cypress.io recordings. We exceeded the free limit for open source projects.

**Comment by ivelin** [comment]
Down to 3 failing assertions for no-pwa presets.

##[error]15 results for https://ui.ambianic.ai/
Report: https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1589388960517-61821.report.html

❌ `font-display` failure for `minScore` assertion (Ensure text remains visible during webfont load: https://web.dev/font-display)
Expected >= 1, but found 0

❌ `unused-css-rules` failure for `maxLength` assertion (Remove unused CSS: https://web.dev/unused-css-rules)
Expected <= 0, but found 3

❌ `uses-rel-preconnect` failure for `maxLength` assertion (Preconnect to required origins: https://web.dev/uses-rel-preconnect)
Expected <= 0, but found 2

**Comment by sniperadmin** [comment]
this looks much better! the report helps alot
so, we will rely on this report for `https://ui.ambianic.ai/` , right?

**Comment by ivelin** [comment]
> this looks much better! the report helps alot
> so, we will rely on this report for `https://ui.ambianic.ai/` , right?

That's what I hope, but still tinkering with the details. Trying to see why LHCI reports different results between two identical Netlify deployments. The report against ui.ambianic.ai should be the same as the report against the PR branch, since there is no app code change.

Then we need to understand why the LHCI report is completely different between a netlify deployment and a local build dist run.

**Comment by ivelin** [comment]
Another day gone in the battle with lhci. Inconsistent results between 3 runs for one partccular assertion:

❌ `font-display` failure for `minScore` assertion (Ensure text remains visible during webfont load: https://web.dev/font-display)
Expected >= 1, but found 0

1. lhci on local dist build - no errors
2. lhci on netlify deployment - multiple errors
3. manual lighthouse chrome on netlify deployment - no errors

Lots of discussions in the public forums and varied answers. It is clear that lhci is early in its product cycle and still maturing. Will give it another push and if I can't figure out what's wrong , will have to postpone until lhci's next stable version.

The cypress e2e testing experience is far more consistent so far as compared to lhci.

**Comment by ivelin** [comment]
@sniperadmin I would like to request a peer review.

After fiddling with lhci for several days, I came to the conclusion that we'd have to have some patience with it while it moves from 0.3 to a stable release.

I ended up cutting off LHCI assertions that show inconsistent results between test runs as described above. I figure we start from a low threshold that's green and stable and gradually build from there. It is better than nothing. Although far from giving us peace of mind that PRs won't introduce PWA regressions.

Let me know.

File: .github/lighthouse/lighthouserc-netlify-prod.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (sniperadmin) [inline .github/lighthouse/lighthouserc-netlify-prod.json:1]: I agree with the split .. things are going longer and longer

**Comment by ivelin** [comment]
@sniperadmin  Thank you for the prompt review and sorry for taking awhile to grog the complexity of using lhci. Merging now. 

Would you be able to update #365 with these changes so we can bring it in next?

**Pull request merged**